app/gui/view/component-browser/breadcrumbs/src/entry.rs

```diff
@@ -0,0 +1,373 @@
+        }
+        let ellipsis = ellipsis::View::new(&logger);
+        let separator = separator::View::new(&logger);
+        let state = default();
+        display_object.add_child(&ellipsis);
+        Self { display_object, state, text, ellipsis, separator }
+    }
+
+    fn remove_current(&self) {
+        match self.state.get() {
+            State::Text => self.text.unset_parent(),
+            State::Separator => self.separator.unset_parent(),
+            State::Ellipsis => self.ellipsis.unset_parent(),
+        }
+    }
+
+    fn set_model(&self, model: &Model) {
+        match model {
+            Model::Text(content) => self.switch_to_text(content),
+            Model::Separator => self.switch_to_separator(),
+            Model::Ellipsis => self.switch_to_ellipsis(),
+        }
+    }
+
+    fn switch_to_text(&self, content: &str) {
+        self.text.set_content(content);
+        if self.state.get() != State::Text {
+            self.remove_current();
+            self.display_object.add_child(&self.text);
+            self.state.set(State::Text);
+        }
+    }
+
+    fn switch_to_separator(&self) {
+        if self.state.get() != State::Separator {
+            self.remove_current();
+            self.display_object.add_child(&self.separator);
+            self.state.set(State::Separator);
+        }
+    }
+
+    fn switch_to_ellipsis(&self) {
+        if self.state.get() != State::Ellipsis {
+            self.remove_current();
+            self.display_object.add_child(&self.ellipsis);
+            self.state.set(State::Ellipsis);
+        }
+    }
+
+    fn update_layout(&self, contour: Contour, text_size: text::Size, text_offset: f32) {
+        let size = contour.size;
+        self.text.set_position_xy(Vector2(text_offset - size.x / 2.0, text_size.raw / 2.0));
+        self.separator.size.set(size);
+        self.ellipsis.size.set(size);
+    }
+
+    fn set_default_color(&self, color: color::Rgba) {
+        self.text.set_default_color(color);
+        self.ellipsis.alpha.set(color.alpha);
+        self.separator.color.set(color.into());
+    }
+
+    fn set_font(&self, font: String) {
+        self.text.set_font(font);
+    }
+
+    fn set_default_text_size(&self, size: text::Size) {
+        self.text.set_default_text_size(size);
+    }
+
+    fn is_state_switch(&self, model: &Model) -> bool {
+        match model {
+            Model::Text(_) => self.state.get() != State::Text,
+            Model::Separator => self.state.get() != State::Separator,
+            Model::Ellipsis => self.state.get() != State::Ellipsis,
+        }
+    }
+
+    fn width(&self, text_offset: f32) -> f32 {
+        match self.state.get() {
+            State::Text => self.text.width.value() + text_offset * 2.0,
+            State::Separator => separator::ICON_WIDTH,
+            State::Ellipsis => ellipsis::ICON_WIDTH,
+        }
+    }
+}
+
+// === Params ===
+
+/// The parameters of Breadcrumbs' entries.
+#[allow(missing_docs)]
```

Comment:
It states that docs are not needed, but in reality they are (or better field names):
1. I dont understand what is "margin".
2. What is `greyed_out_start` ? What is the `usize` there? 
3. What is `text_offset` - offset to what?
4. What is `font`? Is it real font data or is it just font name? We are using "font" var name in many places to always indicate font, not font name.

Reply:
1. Added docs
2. Added docs, replaced `usize` with `Col` (nice catch!)
3. `text_padding_left`
4. We use types, for this reason, don't we? For example, the `DEFAULT_FONT` constant is named as it is, but it should be named `DEFAULT_FONT_NAME` then. Anyway, I renamed it to `font_name`.

Reply:
yup, its ould be `DEFAULT_FONT_NAME`. I believe I changed it in my newest PR (not merged yet).